src/rust/engine/process_execution/src/remote.rs

```diff
@@ -1341,6 +1345,7 @@ pub async fn check_action_cache(
   action_cache_client: Arc<ActionCacheClient<LayeredService>>,
   store: Store,
   eager_fetch: bool,
+  timeout_duration: Option<Duration>,
```

Comment:
Why an `Option`? Looks like you set it everywhere, including `process_executor/main.rs`. Premature generalization?

Reply:
initial draft syndrome -- will remove the Option

Reply:
I've never heard that phrase, I like that :)

Reply:
Since I just invented the term ...